src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationSetting.cs

```diff
@@ -23,17 +22,19 @@ public ConfigurationSetting(string key, string value, string label = null)
             Label = label;
         }
 
+        public ConfigurationSetting(string key, string value, string label, string contentType = null, ETag eTag = default, DateTimeOffset? lastModified = null, bool? locked = null) : this(key, value, label)
```

Comment:
Yeah, I really don't like this. It complicates the mainline API. In particular I don't like that now ETag is settable. I would prefer that such APIs shipped in either a test specific assembly, or at least test namespace/type. In the worst case, we should consider making the APIs not visible in intellisense by either making the setters protected, or applying the visibility attributes